### PR TITLE
feat: expose sourceRepo, sourceRepoOrg and sourceRepoName

### PR DESCRIPTION
**Pull request by elmaso1620**
## Changes:

Additional config variables sourceRepoOrg and sourceRepoName are exposed.

## Context:

This exposes the repository organisation and name so that they can be used in PR notes:
https://github.com/renovatebot/renovate/discussions/13601

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/updates/flatten.ts [hunk 1]
```diff
@@ -42,6 +42,14 @@ export function applyUpdateConfig(input: BranchUpgradeConfig): any {
         .replace(regEx(/^\//), '') // remove leading slash
         .replace(regEx(/\//g), '-') // change slashes to hyphens
         .replace(regEx(/-+/g), '-'); // remove multiple hyphens
+      updateConfig.sourceRepoOrg = parsedSourceUrl.pathname
+        .replace(regEx(/^\//), '') // remove leading slash
+        .replace(regEx(/\/.*/g), '') // remove everything after first slash
+        .replace(regEx(/-+/g), '-'); // remove multiple hyphens
```
Comment (viceice) [inline lib/workers/repository/updates/flatten.ts:48]:
```suggestion
        .replace(regEx(/\/.*/g), ''); // remove everything after first slash
```
remove hyphens will invalidate org

File: lib/workers/repository/updates/flatten.ts [hunk 2]
```diff
@@ -42,6 +42,14 @@ export function applyUpdateConfig(input: BranchUpgradeConfig): any {
...
+      updateConfig.sourceRepoName = parsedSourceUrl.pathname
+        .replace(regEx(/^\//), '') // remove leading slash
+        .replace(regEx(/^(.*?)\//g), '') // remove everything up to first slash
+        .replace(regEx(/-+/g), '-'); // remove multiple hyphens
```
Comment (viceice) [inline lib/workers/repository/updates/flatten.ts:52]:
```suggestion
        .replace(regEx(/^(.*?)\//g), ''); // remove everything up to first slash
```

same here.

I think you need to split at last slash, as on gitlab we cvan have subgroups.

`group/sub/subsub/repo` => `sourceRepoOrg=group/sub/subsub` and `sourceRepoName=repo`

**Review by viceice — COMMENTED** [review]
Please sign CLA

File: lib/workers/repository/updates/flatten.spec.ts [hunk 3]
```diff
@@ -143,6 +143,8 @@ describe('workers/repository/updates/flatten', () => {
       const res = await flattenUpdates(config, packageFiles);
       expect(res).toHaveLength(14);
       expect(res.filter((update) => update.sourceRepoSlug)).toHaveLength(3);
+      expect(res.filter((update) => update.sourceRepoOrg)).toHaveLength(3);
+      expect(res.filter((update) => update.sourceRepoName)).toHaveLength(3);
```
Comment (viceice) [inline lib/workers/repository/updates/flatten.spec.ts:147]: i think this should have some more explicit expects, so it will work as expected.

**Review by viceice — COMMENTED** [review]
I think you now need to extend `allowedFields`, so the new values are allowed in templating

https://github.com/renovatebot/renovate/blob/77708881845c9520e29e878db66830d88a5467c5/lib/util/template/index.ts#L66

**Comment by elmaso1620** [comment]
> I think you now need to extend `allowedFields`, so the new values are allowed in templating
> 
> https://github.com/renovatebot/renovate/blob/77708881845c9520e29e878db66830d88a5467c5/lib/util/template/index.ts#L66

Thanks for you review comments so far but I don't understand this last one about extending 'allowedFields'. I've already added sourceRepoOrg and sourceRepoName into the allowedFields list: https://github.com/elmaso1620/renovate/blob/eec0def95b537b8e631cbc45b42e041b293310f2/lib/util/template/index.ts#L119

File: lib/util/template/index.ts [hunk 4]
```diff
@@ -116,6 +116,8 @@ export const allowedFields = {
   releaseNotes: 'A ChangeLogNotes object for the release',
   repository: 'The current repository',
   semanticPrefix: 'The fully generated semantic prefix for commit messages',
+  sourceRepoName: 'The repository name in the sourceUrl, if present',
+  sourceRepoOrg: 'The repository organization in the sourceUrl, if present',
   sourceRepoSlug: 'The slugified pathname of the sourceUrl, if present',
```
Comment (rarkins) [inline lib/util/template/index.ts:121]: Are we missing `sourceRepo` here? e.g. `foo/bar`? Would be nice to add now if so

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/repository/updates/flatten.ts [hunk 5]
```diff
@@ -42,6 +42,16 @@ export function applyUpdateConfig(input: BranchUpgradeConfig): any {
         .replace(regEx(/^\//), '') // remove leading slash
         .replace(regEx(/\//g), '-') // change slashes to hyphens
         .replace(regEx(/-+/g), '-'); // remove multiple hyphens
+      updateConfig.sourceRepo = parsedSourceUrl.pathname.replace(
+        regEx(/^\//),
+        ''
+      ); // remove leading slash
+      updateConfig.sourceRepoOrg = parsedSourceUrl.pathname
+        .replace(regEx(/^\//), '') // remove leading slash
```
Comment (rarkins) [inline lib/workers/repository/updates/flatten.ts:50]: Wouldn't the `sourceRepo` also benefit from removing any leading slash?

Comment (elmaso1620) [reply lib/workers/repository/updates/flatten.ts:50]: The leading slashes for sourceRepo are removed here: https://github.com/elmaso1620/renovate/blob/564b87da4b28f898219b6e0e50210969d0d18f73/lib/workers/repository/updates/flatten.ts#L48

Comment (rarkins) [reply lib/workers/repository/updates/flatten.ts:50]: OK, changing my recommendation with a new comment

Comment (rarkins) [inline lib/workers/repository/updates/flatten.ts:50]:
```suggestion
      updateConfig.sourceRepoOrg = updateConfig.sourceRepo
```

File: lib/workers/repository/updates/flatten.ts [hunk 6]
```diff
@@ -42,6 +42,16 @@ export function applyUpdateConfig(input: BranchUpgradeConfig): any {
...
+        .replace(regEx(/\/.*/g), ''); // remove everything after first slash
+      updateConfig.sourceRepoName = parsedSourceUrl.pathname
+        .replace(regEx(/^\//), '') // remove leading slash
```
Comment (rarkins) [inline lib/workers/repository/updates/flatten.ts:53]:
```suggestion
      updateConfig.sourceRepoName = updateConfig.sourceRepo
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — DISMISSED** [review]

File: lib/workers/repository/updates/flatten.ts [hunk 7]
```diff
@@ -42,6 +42,18 @@ export function applyUpdateConfig(input: BranchUpgradeConfig): any {
         .replace(regEx(/^\//), '') // remove leading slash
         .replace(regEx(/\//g), '-') // change slashes to hyphens
         .replace(regEx(/-+/g), '-'); // remove multiple hyphens
+      updateConfig.sourceRepo = parsedSourceUrl.pathname.replace(
+        regEx(/^\//),
+        ''
+      ); // remove leading slash
+      updateConfig.sourceRepoOrg = updateConfig.sourceRepo.replace(
```
Comment (viceice) [inline lib/workers/repository/updates/flatten.ts:49]: Something seems wrong now

**Review by viceice — CHANGES_REQUESTED** [review]
Failing tests

File: lib/workers/repository/updates/flatten.ts (hunk 7 above)
Comment (viceice) [reply lib/workers/repository/updates/flatten.ts:49]: You propably need to declare the fields on the interface

File: lib/workers/types.ts [hunk 8]
```diff
@@ -69,6 +69,7 @@ export interface BranchUpgradeConfig
   changelogUrl?: string;
   dependencyUrl?: string;
   sourceUrl?: string;
+  sourceRepo?: string;
```
Comment (viceice) [inline lib/workers/types.ts:72]:
```suggestion
  sourceRepo?: string;
  sourceRepoOrg?: string;
  sourceRepoName?: string;
```

We should add all new fields 😉

**Review by viceice — APPROVED** [review]

**Pull request merged**